astronomer/providers/utils/sensor_util.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from airflow.exceptions import AirflowException, AirflowSkipException
+
+if TYPE_CHECKING:  # pragma: no cover
+    from astronomer.providers.utils.typing_compat import Context
+
+
+def poke(cls: Any, context: Context) -> bool | None:  # pragma: no cover
```

Comment:
Also, I'm wondering why we need this `poke` function. In what scenario will `poke` raise an exception?

Reply:
This is a wrapper to call the upstream poke method. We call poke method in almost all sensors before deferring so we need to handle the error if any occurs. This  poke method helps to avoid writing try-catch at  multiple places

Reply:
> In what scenario will poke raise an exception?

maybe a call to s3 and call fail either network or something else

Reply:
Got it. Sounds good 👍